sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/aad/AADAuthenticationProperties.java

```diff
@@ -139,23 +141,48 @@
          * Expected UserGroups that an authority will be granted to if found in the response from the MemeberOf Graph
          * API Call.
          */
-        private List<String> allowedGroups = new ArrayList<>();
+        private List<String> allowedGroupNames = new ArrayList<>();
 
+        private Set<String> allowedGroupIds = new HashSet<>();
+
+        public Set<String> getAllowedGroupIds() {
+            return allowedGroupIds;
+        }
+
+        public void setAllowedGroupIds(Set<String> allowedGroupIds) {
+            this.allowedGroupIds = allowedGroupIds;
+        }
+
+        public List<String> getAllowedGroupNames() {
+            return allowedGroupNames;
+        }
+
+        public void setAllowedGroupNames(List<String> allowedGroupNames) {
+            this.allowedGroupNames = allowedGroupNames;
+        }
+
+        @Deprecated
         public List<String> getAllowedGroups() {
-            return allowedGroups;
+            return allowedGroupNames;
         }
 
+        @Deprecated
         public void setAllowedGroups(List<String> allowedGroups) {
-            this.allowedGroups = allowedGroups;
+            this.allowedGroupNames = allowedGroups;
         }
 
     }
 
     public boolean allowedGroupsConfigured() {
         return Optional.of(this)
                        .map(AADAuthenticationProperties::getUserGroup)
-                       .map(AADAuthenticationProperties.UserGroupProperties::getAllowedGroups)
-                       .map(allowedGroups -> !allowedGroups.isEmpty())
+                       .map(AADAuthenticationProperties.UserGroupProperties::getAllowedGroupNames)
+                       .map(filterByNames -> !filterByNames.isEmpty())
+                       .orElse(false)
+            || Optional.of(this)
```

Comment:
this code looks really wired, would better to refactor

* Optional.of(this) ???
* why do you need to use a "||"?

Reply:
- Refactoring it will make it easier to understand.
- Add "| |" in order to determine whether group name or group id is allowed, it will check to see if the two Collection contains.